utils/genesis/src/lib.rs

```diff
@@ -70,6 +77,15 @@ where
     Ok(network_name)
 }
 
+pub async fn get_network_name_from_genesis_cid(cid: &str) -> Result<String, anyhow::Error> {
+    let network_name = match cid {
+        MAINNET_GENESIS_CID => "mainnet".to_owned(),
+        CALIBNET_GENESIS_CID => "calibnet".to_owned(),
+        _ => bail!("Cannot guess network name using cid: {}", cid),
```

Comment:
We want to support custom networks (ie local dev networks) in the future. How about we make the expected genesis CID part of the `ChainConfig`? That way, a custom configuration can be used together with a custom network.

Reply:
done